packages/editor/src/dataviews/types.ts

```diff
@@ -0,0 +1,21 @@
+type PostStatus =
+	| 'published'
+	| 'draft'
+	| 'pending'
+	| 'private'
+	| 'future'
+	| 'auto-draft'
+	| 'trash';
+
+export interface BasePost {
+	status?: PostStatus;
+	title: string | { rendered: string };
+	type: string;
+}
+export interface TemplateOrTemplatePart extends BasePost {
+	type: 'template' | 'template-part';
+	source: string;
+	has_theme_file: boolean;
+}
+
+export type Post = TemplateOrTemplatePart | BasePost;
```

Comment:
I like a lot that we could use the "actions" refactoring to slowly augment our "entities" types. In the long run, this will help a lot when build UI for entities.